### PR TITLE
copyDesktopItems: Use bin output

### PR DESCRIPTION
**Pull request by toonn**
###### Description of changes

Desktop files are only useful when accompanied by the binaries specify. So it makes more sense to put them next to the binaries than `$out` which only usually contains the binaries.

This may affect packages that have worked around this behavior by manually copying things to `$out/share` rather than next to the binaries. Not sure what the best way is to smoke these out.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/setup-hooks/copy-desktop-items.sh [hunk 1]
```diff
@@ -22,21 +22,21 @@
 postInstallHooks+=(copyDesktopItems)
 
 copyDesktopItems() {
-    if [ "${dontCopyDesktopItems-}" = 1 ]; then return; fi
+  if [ "${dontCopyDesktopItems-}" = 1 ]; then return; fi
```
Comment (SuperSandro2000) [inline pkgs/build-support/setup-hooks/copy-desktop-items.sh:25]: shell files are supposed to have 4 spaces, please revert.

Comment (SuperSandro2000) [reply pkgs/build-support/setup-hooks/copy-desktop-items.sh]: I don't think we have a line limit other than not-way-to-long.

File: pkgs/build-support/setup-hooks/copy-desktop-items.sh [hunk 2]
```diff
@@ -30,12 +30,14 @@ copyDesktopItems() {
 
     for desktopItem in $desktopItems; do
         if [[ -f "$desktopItem" ]]; then
-            echo "Copying '$desktopItem' into '$out/share/applications'"
-            install -D -m 444 -t "$out"/share/applications "$desktopItem"
+            echo \
+              "Copying '$desktopItem' into '${!outputBin}/share/applications'"
```
Comment (K900) [inline pkgs/build-support/setup-hooks/copy-desktop-items.sh:34]: Maybe stick this in a variable to avoid repeating it everywhere?

Comment (toonn) [reply pkgs/build-support/setup-hooks/copy-desktop-items.sh:34]: Definitely cleaner 👍

**Review by K900 — APPROVED** [review]
LGTM

**Comment by Artturin** [comment]
Result of `nixpkgs-review pr 163516` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>deltachat-desktop</li>
    <li>pomotroid</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>ciscoPacketTracer7</li>
    <li>ciscoPacketTracer8</li>
    <li>openboard</li>
    <li>snapdragon-profiler</li>
  </ul>
</details>
<details>
  <summary>189 packages built:</summary>
  <ul>
    <li>abuse</li>
    <li>ace-of-penguins</li>
    <li>aether</li>
    <li>agi</li>
    <li>antimony</li>
    <li>appflowy</li>
    <li>azuredatastudio</li>
    <li>ballerburg</li>
    <li>basex</li>
    <li>berry</li>
    <li>bisq-desktop</li>
    <li>compcert</li>
    <li>copyDesktopItems</li>
    <li>coq (coqPackages.coq ,coq_8_13)</li>
    <li>coqPackages.Cheerios</li>
    <li>coqPackages.CoLoR</li>
    <li>coqPackages.ITree</li>
    <li>coqPackages.InfSeqExt</li>
    <li>coqPackages.LibHyps</li>
    <li>coqPackages.QuickChick</li>
    <li>coqPackages.StructTact</li>
    <li>coqPackages.VST</li>
    <li>coqPackages.Verdi</li>
    <li>coqPackages.aac-tactics</li>
    <li>coqPackages.addition-chains</li>
    <li>coqPackages.autosubst</li>
    <li>coqPackages.bignums</li>
    <li>coqPackages.category-theory</li>
    <li>coqPackages.ceres</li>
    <li>coqPackages.compcert</li>
    <li>coqPackages.coq-bits</li>
    <li>coqPackages.coq-elpi</li>
    <li>coqPackages.coq-ext-lib</li>
    <li>coqPackages.coq-record-update</li>
    <li>coqPackages.coqeal</li>
    <li>coqPackages.coqhammer</li>
    <li>coqPackages.coqprime</li>
    <li>coqPackages.coqtail-math</li>
    <li>coqPackages.coquelicot</li>
    <li>coqPackages.corn</li>
    <li>coqPackages.deriving</li>
    <li>coqPackages.dpdgraph</li>
    <li>coqPackages.equations</li>
    <li>coqPackages.extructures</li>
    <li>coqPackages.flocq</li>
    <li>coqPackages.fourcolor</li>
    <li>coqPackages.gaia</li>
    <li>coqPackages.gaia-hydras</li>
    <li>coqPackages.gappalib</li>
    <li>coqPackages.goedel</li>
    <li>coqPackages.graph-theory</li>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.hydra-battles</li>
    <li>coqPackages.interval</li>
    <li>coqPackages.iris</li>
    <li>coqPackages.itauto</li>
    <li>coqPackages.math-classes</li>
    <li>coqPackages.mathcomp</li>
    <li>coqPackages.mathcomp-abel</li>
    <li>coqPackages.mathcomp-algebra</li>
    <li>coqPackages.mathcomp-analysis</li>
    <li>coqPackages.mathcomp-bigenough</li>
    <li>coqPackages.mathcomp-character</li>
    <li>coqPackages.mathcomp-field</li>
    <li>coqPackages.mathcomp-fingroup</li>
    <li>coqPackages.mathcomp-finmap</li>
    <li>coqPackages.mathcomp-real-closed</li>
    <li>coqPackages.mathcomp-solvable</li>
    <li>coqPackages.ssreflect (coqPackages.mathcomp-ssreflect)</li>
    <li>coqPackages.mathcomp-tarjan</li>
    <li>coqPackages.mathcomp-word</li>
    <li>coqPackages.mathcomp-zify</li>
    <li>coqPackages.metalib</li>
    <li>coqPackages.multinomials</li>
    <li>coqPackages.odd-order</li>
    <li>coqPackages.paco</li>
    <li>coqPackages.paramcoq</li>
    <li>coqPackages.parsec</li>
    <li>coqPackages.pocklington</li>
    <li>coqPackages.reglang</li>
    <li>coqPackages.relation-algebra</li>
    <li>coqPackages.semantics</li>
    <li>coqPackages.serapi</li>
    <li>coqPackages.simple-io</li>
    <li>coqPackages.smpl</li>
    <li>coqPackages.stdpp</li>
    <li>coqPackages.tlc</li>
    <li>coqPackages.topology</li>
    <li>coqPackages.zorns-lemma</li>
    <li>coq_8_10</li>
    <li>coq_8_11</li>
    <li>coq_8_12</li>
    <li>coq_8_14</li>
    <li>coq_8_15</li>
    <li>coq_8_5</li>
    <li>coq_8_6</li>
    <li>coq_8_7</li>
    <li>coq_8_8</li>
    <li>coq_8_9</li>
    <li>cubiomes-viewer</li>
    <li>dbeaver</li>
    <li>dhewm3</li>
    <li>dirdiff</li>
    <li>dosbox</li>
    <li>dosbox-staging</li>
    <li>ecwolf</li>
    <li>emuflight-configurator</li>
    <li>fiji</li>
    <li>fish-fillets-ng</li>
    <li>framac</li>
    <li>free42</li>
    <li>freedroid</li>
    <li>freeoffice</li>
    <li>frozen-bubble</li>
    <li>gaphor</li>
    <li>gitkraken</li>
    <li>gkrellm</li>
    <li>gnujump</li>
    <li>goattracker</li>
    <li>goattracker-stereo</li>
    <li>gprojector</li>
    <li>gpsprune</li>
    <li>gzdoom</li>
    <li>hdfview</li>
    <li>hdrmerge</li>
    <li>imagej</li>
    <li>inav-configurator</li>
    <li>indigenous-desktop</li>
    <li>ipe</li>
    <li>irpf</li>
    <li>jd-gui</li>
    <li>jdiskreport</li>
    <li>jquake</li>
    <li>keen4</li>
    <li>key</li>
    <li>lapce</li>
    <li>ligo</li>
    <li>logisim</li>
    <li>logisim-evolution</li>
    <li>lsd2dsl</li>
    <li>mars-mips</li>
    <li>mikutter</li>
    <li>mindustry</li>
    <li>mindustry-wayland</li>
    <li>minecraft</li>
    <li>noisetorch</li>
    <li>oh-my-git</li>
    <li>open-watcom-v2</li>
    <li>open-watcom-v2-unwrapped</li>
    <li>opensupaplex</li>
    <li>opentabletdriver</li>
    <li>opentrack</li>
    <li>osu-lazer</li>
    <li>pdfstudio</li>
    <li>pdfstudioviewer</li>
    <li>premid</li>
    <li>protege-distribution</li>
    <li>prusa-slicer</li>
    <li>pyspread</li>
    <li>quakespasm</li>
    <li>rare</li>
    <li>rott</li>
    <li>rott-shareware</li>
    <li>rstudio</li>
    <li>rstudioWrapper</li>
    <li>rustdesk</li>
    <li>satallax</li>
    <li>schildichat-desktop</li>
    <li>schildichat-desktop-wayland</li>
    <li>sdlpop</li>
    <li>sharedown</li>
    <li>snapmaker-luban</li>
    <li>softmaker-office</li>
    <li>sony-headphones-client</li>
    <li>starsector</li>
    <li>stm32cubemx</li>
    <li>super-slicer (super-slicer-latest)</li>
    <li>thonny</li>
    <li>timeline</li>
    <li>trenchbroom</li>
    <li>tutanota-desktop</li>
    <li>unciv</li>
    <li>unison</li>
    <li>unvanquished</li>
    <li>usync</li>
    <li>vintagestory</li>
    <li>why3</li>
    <li>xbill</li>
    <li>xcolor</li>
  </ul>
</details>

**Comment by toonn** [comment]
@Artturin, what was the reason for the `openboard` failure? It doesn't seem to use `requireFile`.

**Comment by Artturin** [comment]
same as on hydra https://hydra.nixos.org/build/171891455

**Pull request merged**